docs/docs/data-structures/tree/schema-evolution/index.mdx

```diff
@@ -0,0 +1,219 @@
+    console.log("Document schema was updated by another client");
+});
+```
+
+## Schema Upgrade Process
+
+When you make [backwards compatible](./index.mdx#understanding-compatibility) changes to your schema, you can upgrade existing documents to use the new schema using [`view.upgradeSchema()`](../../../api/fluid-framework/treeview-interface#upgradeschema-method):
+
+```typescript
+import { SchemaFactory, TreeViewConfiguration } from "@fluidframework/tree";
+
+const factory = new SchemaFactory("WhiteboardApp");
+
+// Updated schema with new optional field
+class Note extends factory.object("Note", {
+    id: factory.string,
+    x: factory.number,
+    y: factory.number,
+    color: factory.optional(factory.string), // New optional field
+}) {}
+
+// Create tree view with updated schema
+const config = new TreeViewConfiguration({ schema: Note });
+const view = tree.viewWith(config);
+
+// Check compatibility and upgrade if needed
+if (!view.compatibility.isEquivalent) {
+    if (view.compatibility.canUpgrade) {
+        view.upgradeSchema(); // Upgrade document to new schema
+        console.log("Document upgraded successfully");
+    } else {
+        throw new Error("Schema is incompatible and cannot be upgraded");
+    }
+}
+
+// Now you can use the upgraded document
+view.root.color = "#FF0000";
+```
+
+For detailed information about which types of changes are safe versus breaking, see [Types of Schema Changes](./types-of-changes).
+
+### Staged Rollouts
+
+When deploying schema changes in production, use staged rollouts to ensure compatibility between clients running different application versions.
+
+**Why Staged Rollouts Are Necessary**
+
+In collaborative applications, multiple users with different client versions may work on the same document simultaneously.
+If a client upgrades a document to a new schema that older clients cannot read, those users will be locked out until they update their application.
+
+**Two-Phase Deployment Strategy**
+
+**Phase 1: Deploy Schema Reading Support**
+- Deploy application versions that can read the new schema without upgrading documents
+- Wait until all (or "tolerably most") users have updated to the new application version. This is called **client saturation**.
+- Monitor deployment metrics to ensure coverage
+
+**Phase 2: Enable Document Upgrades**
+- Deploy code that automatically upgrades documents to the new schema
+- All clients can now collaborate since they support reading the upgraded schema
+- Monitor for any compatibility issues
+
+Theoretically, some changes don't require a staged rollout and are therefore capable of cross-version collaboration.
+In that case, new clients can read and write to the same document as old clients without either one being fundamentally incompatible.
+However, staged rollouts are not easily avoidable. Even seemingly simple schema changes, like adding a single required field to a node, cannot support cross-version collaboration and require a staged rollout.
```

Comment:
```suggestion
However, staged rollouts are not easily avoidable. Even seemingly simple schema changes (for example, adding a new type of node to a field) may not be forwards-compatible, cannot support cross-version collaboration and therefore require a staged rollout.
```

Adding required fields is a confusing example because it's not possible except it is possible in a roundabout way. Probably a detail to be saved for the type-of-changes document.